### ISSUE TITLE
Different output for eslint when run from NPM Script on Windows

### ISSUE THREAD

**Issue by gauntface** [eslint/eslint#9306]
**Tell us about your environment**

* **ESLint Version:** 4.6.1
* **Node Version:** 8.5.0
* **npm Version:** 5.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
None.

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: ['eslint:recommended', 'google'],
  env: {
    serviceworker: true,
    browser: true,
    node: true,
    es6: true,
  },
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'module',
  },
  rules: {
  },
  overrides: [{
    files: ['test/**/*.{js,mjs}'],
    env: {
      mocha: true,
    },
    rules: {
      'valid-jsdoc': 0,
      'require-jsdoc': 0,
      'max-len': 0,
    },
  }, {
    files: [
      'packages/workbox-core/src/utils/LogHelper.js',
      'test/workbox-core/bundle/node/utils/test-LogHelper.js',
      'infra/utils/log-helper.js',
      'infra/tools/analyse-properties.js',
    ],
    rules: {
      'no-console': 0,
    },
  }, {
    files: [
      'gulp-tasks/**/*.js',
      'infra/**/*.js'
    ],
    rules: {
      'valid-jsdoc': 0,
      'require-jsdoc': 0,
    },
  }, {
    files: [
      'infra/utils/generate-variant-tests.js',
    ],
    env: {
      'mocha': true
    }
  }],
};
```

</details>

**What did you do? Please include the actual source code causing the issue.**

It's just line endings on a windows box and an unused variable.

**What did you expect to happen?**

When running eslint via npm run script `npm run lint` to have the same output as `node .\node_modules\eslint\bin\eslint.js`.

**What actually happened? Please include the actual, raw output from ESLint.**
When I run from node:

```
PS C:\Users\mattt_000\projects\workbox> node .\node_modules\eslint\bin\eslint.js '.' --ignore-path .gitignore

C:\Users\mattt_000\projects\workbox\gulp-tasks\utils\spawn-promise-wrapper.js
   1:46  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   2:1   error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   3:52  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   4:46  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   5:44  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   6:51  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   7:33  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   8:36  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
   9:24  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
  10:19  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
  11:15  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
  12:66  error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
  13:8   error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
  14:8   error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
  15:6   error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style
  16:3   error  Expected linebreaks to be 'LF' but found 'CRLF'  linebreak-style

✖ 16 problems (16 errors, 0 warnings)
  16 errors, 0 warnings potentially fixable with the `--fix` option.

```

When I run from npm script

```
PS C:\Users\mattt_000\projects\workbox> npm run lint

> workbox@0.0.0 lint C:\Users\mattt_000\projects\workbox
> eslint '.' --ignore-path .gitignore
```
I'm only seeing this behaviour on Windows.

**Comment by not-an-aardvark** [comment]
Thanks for the report. I'm not quite sure why that's happening, although if anything I think it would an issue with npm, not ESLint.

What happens if you change the npm script to target `gulp-tasks\utils\spawn-promise-wrapper.js` explicitly? Maybe your shell is doing something with the `.` path before it gets passed to ESLint.

**Comment by mysticatea** [comment]
You cannot use single quotes to enclose file path in npm scripts. Please use double quotes instead.

- `powershell.exe` strips the single quotes as you expected.
- `cmd.exe` does not strip the single quotes. The single quotes are parts of the file path, so `'.'` is not the current directory.
- `npm run-script` uses `cmd.exe` by design of `mpm`.
